### ISSUE TITLE
[Question] BC-RNN policy oscillates near target object without grasping during evaluation

### ISSUE THREAD

**Issue by thiendoengr-svg** [isaac-sim/IsaacLab#7963]
### Question

I am encountering an issue where a Policy trained with Behavior Cloning (BC-RNN) fails to execute the grasping task during evaluation (`play.py`). Although the robot arm successfully moves toward the target object's spatial location, it oscillates around the object without executing the final pick/grasp action, despite low training loss.

Specifically, I would like to clarify if this issue stems from the BC-RNN network architecture (such as temporal horizon/LSTM state handling), sub-optimal spatial precision from the visual encoder, or action/gripper output scaling during the execution phase.

**Details of the setup**:
1. **Model Architecture:** BC-RNN with ResNet18 backbone (`ResNet18Conv`), `SpatialSoftmax` pooling (32 keypoints), concatenated with low-dim states (`joint_pos`, `joint_vel`, `cube_red_pos`, `last_action`), fed into a 2-layer LSTM (`hidden_dim: 256`, `sequence_length: 10`).
2. **Recorded Dataset:** HDF5 dataset containing dual RGB camera streams (`workspace_cam`, `global_cam`) and low-dimensional states (`joint_pos`, `joint_vel`, `cube_red_pos`, `last_action`).
3. **Training Behavior:** BC L2 loss decreases smoothly across 600 epochs. However, during evaluation (`play.py`), the robot only reaches toward the object's position distribution and hovers/oscillates around it without gripping or completing the task.

**Code & Data Link:**
My configuration file and training setup can be found here: https://github.com/thiendoengr-svg/bc_training_so101

**Comment by thiendoengr-svg** [comment]
Train Epoch 1
{
    "Cosine_Loss": 0.1448293970171362,
    "L1_Loss": 0.016693633393384515,
    "L2_Loss": 0.033545225797221065,
    "Loss": 0.033545225797221065,
    "Optimizer/policy0_lr": 0.0003,
    "Policy_Grad_Norms": 0.05198244140539887,
    "Time_Data_Loading": 0.27344295581181843,
    "Time_Epoch": 3.308647878964742,
    "Time_Log_Info": 0.013441340128580729,
    "Time_Process_Batch": 0.4596757968266805,
    "Time_Train_Batch": 2.5572218457857767
}
100%|##########| 10/10 [00:21<00:00,  2.17s/it]
Validation Epoch 1
{
    "Cosine_Loss": 0.0738556120544672,
    "L1_Loss": 0.0074143156874924895,
    "L2_Loss": 0.014828631374984979,
    "Loss": 0.014828631374984979,
    "Optimizer/policy0_lr": 0.0003,
    "Time_Data_Loading": 0.33826125065485635,
    "Time_Epoch": 0.36187443335851033,
    "Time_Log_Info": 8.928775787353516e-06,
    "Time_Process_Batch": 0.006571849187215169,
    "Time_Train_Batch": 0.016334942976633706
}
sa
